fenix/app/src/main/java/org/mozilla/fenix/session/VisibilityLifecycleCallback.kt

```diff
@@ -62,6 +67,23 @@ class VisibilityLifecycleCallback(private val activityManager: ActivityManager?)
 
     override fun onActivityDestroyed(activity: Activity) {}
 
+    /**
+     * For private tabs, set the tracking protection exception
+     * to the default state if the app is closed.
+     */
+    private fun removeTCPException(activity: Activity) {
+        val privateTabs = activity.components.core.store.state.privateTabs
+        for (privateTab in privateTabs) {
+            val trackingProtectionUseCases =
+                activity.components.useCases.trackingProtectionUseCases
+            trackingProtectionUseCases.containsException(privateTab.id) {
+                if (it) {
+                    trackingProtectionUseCases.removeException(privateTab.id)
+                }
+            }
```

Comment:
As this is doing a round trip to gecko, we could avoid doing that by query `tab.trackingProtection.ignoredOnTrackingProtection` which is set every time we [add/remove](https://github.com/mozilla-mobile/firefox-android/blob/d88fcfa9c409459bf5a7b5e83d56c3bd3c394879/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoTrackingProtectionExceptionStorage.kt#L71) a TP exception for a session.

>         activity.components.core.store.state.privateTabs.filter {
>             it.trackingProtection.ignoredOnTrackingProtection
>         }.forEach { privateTab ->
>             activity.components.useCases.trackingProtectionUseCases.removeException(privateTab.id)
>         }

Reply:
Done